text/0000-formatted-bliss.md

```diff
@@ -0,0 +1,226 @@
+This will ensure that most Rust developers feel at home with style inside the repository.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+See the [summary] for a brief explanation or read below for a detailed technical
+description of what this proposal entails.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+To disrupt normal daily development as little as possible in the repository,
+the proposal has been divided into steps which in part can and should be
+executed independently of each other.
+
+## 1. Introducing *the style*
+
+As noted in the [motivation], we wish to make the repository where any Rust
+developer feels at home. To that end, we introduce and adopt the standard
+Rust Style Guide into the repository.
+
+*The style* is defined in terms of a `rustfmt.toml` file (hence: *the style file*):
+
+```toml
+version = "Two"
+use_small_heuristics = "Max"
+```
+
+The style file is located at the top of the repository.
+
+This step has already been completed.
+
+## 2. Enforce the style
+
+The `tidy` tool, defined in `src/tools/tidy`, is run each time a new PR is created
+or when a commit is added to said PR. The tool is also run when a PR,
+approved for merging with e.g. `@bors r=reviewer`, is tested for merging.
+
+To `tidy`, new logic is added that checks whether the code is conformant with
+the style. This will result in a PR being rejected when changes in said PR does
+not conform to the style.
+
+To facilitate incremental enforcement of the style and to disrupt development
+minimally inside the repository, the `tidy` tool will only enforce the style
+in folders with an `.enforcefmt` file. In such folders, the style will be
+checked for all files in the immediate folder as well as all subfolders.
+
+It is expected that applying the style throughout the repository will generate
+large diffs. Since it is unfeasible to effectively review so many changes,
+to avoid any security risks, we will only accept PRs applying the style by
+regulars who themselves have reviewer rights in the repository.
+
+Eventually, when the entire repository has been formatted,
+the `.enforcefmt` trigger may be removed in favor of checking everything.
+
+## 3. Add the `./x.py fmt` command
+
+A command `fmt` is added to rustbuild (`./x.py`).
+The command will format the repository according to the style
+but only in folders with `.enforcefmt` files as described above.
+
+## 4. Add a `@rustbot fmt` command
+
+Sometimes compiler developers do not have the time, or may simply forget,
+to apply `./x.py fmt`. In other situations, the developer may not have a
+computer with a compiler and git handy. It may also happen that the PR
+is perfectly good aside from formatting. For these situations, a
+`@rustbot fmt` command is provided with which the author, a reviewer,
+or a T-release member can simply ask `@rustbot` to run `./x.py fmt`
+on their PR and add a commit for that.
```

Comment:
cc @pietroalbini @Mark-Simulacrum

Reply:
I don't know when we'll have the time to implement this, but it seems reasonable.

Reply:
@Mark-Simulacrum The one thing I'm wondering is if `@rustbot fmt` can add commits to the branch of a PR author or not.

Reply:
That's definitely possible, at least in general -- the PR author can untick the relevant checkbox but there's no real reason we need to worry about that. The feature is enabled by default.

Reply:
🎉

Reply:
I agree with Mark.